### ISSUE TITLE
Error when processing coverage reports containing a big number of hits

### ISSUE THREAD

**Issue by wenns** [SonarOpenCommunity/sonar-cxx#65]
With in the Coverage analysis module, parsing a "hit" element that exceeds a _int32 size will cause failures, and emits no useful message.
This is crippling on projects with large code base since the hit count for many central functions can rapidly exceed 2^31.
In my case an accurate hit count is not required, so truncation would be acceptable.
03:03:28.915 INFO - Parsing report '<snip>/Sonar_Test_coverage/software/coverage-reports/coverage-unit.xml'
INFO: ----------------------------------------------------------------------
INFO: EXECUTION FAILURE
INFO: ------------------------------------------------------------------------
INFO: Total time: 38:42.225s
INFO: Final Memory: 25M/2087M
INFO: ------------------------------------------------------------------------
ERROR: Error during Sonar runner execution
ERROR: Unable to execute Sonar
ERROR: Caused by: For input string: "2806720497"
ERROR:

**Comment by brunocrt** [comment]
Hi wenns, could you please provide how did you fix this issue?

**Comment by wenns** [comment]
This was fixed by commit fb3aa88b715338f73f3121fd9d7ff22a835bb55b:

```C++
long noHits = Long.parseLong(line.getAttrValue("hits"));
if(noHits > Integer.MAX_VALUE){
        CxxUtils.LOG.warn("Truncating the actual number of hits ({}) to the maximum number supported by Sonar ({})",
                          noHits, Integer.MAX_VALUE);
        noHits = Integer.MAX_VALUE;
}
builder.setHits(lineId, (int)noHits);
```
